### PR TITLE
boris/log_match_name_in_sanctions

### PR DESCRIPTION
**Pull request by bork1n**

### CODE REVIEW

File: lib/Data/Validate/Sanctions.pm [hunk 1]
```diff
@@ -57,13 +57,15 @@ sub is_sanctioned {        ## no critic (RequireArgUnpacking)
 
     for my $k (sort keys %$data) {
         foreach my $name (@{$data->{$k}{names}}) {
+            (my $check_name = $name) =~ s/[[:^alpha:]]//g;
+            $check_name = uc($check_name);
             for (@name_variants) {
-                return $k if $name =~ /$_/;
-        }
+                return wantarray ? ($k, $name) : $k if $check_name =~ /$_/;
```
Comment (binarytom) [inline lib/Data/Validate/Sanctions.pm]: worth a `/\Q$_/` here to make the intent slightly clearer? (I don't think we'd expect to have any regex characters at that point, so it shouldn't affect the functionality)

File: lib/Data/Validate/Sanctions.pm [hunk 2]
```diff
@@ -57,13 +57,15 @@ sub is_sanctioned {        ## no critic (RequireArgUnpacking)
...
+            }
         }
     }
 
-    return 0;
+    return wantarray ? (0, '') : 0;
```
Comment (binarytom) [inline lib/Data/Validate/Sanctions.pm]: I'd suggest against this - might as well stick to scalar if possible?

File: lib/Data/Validate/Sanctions.pm (hunk 1 above)
Comment (bork1n) [inline lib/Data/Validate/Sanctions.pm]: Actually `$_` is regexp here, we have `.*` between names

File: lib/Data/Validate/Sanctions.pm (hunk 2 above)
Comment (bork1n) [inline lib/Data/Validate/Sanctions.pm]: We need two items - list and name. I can make this function to return single string, like `list:name`. Does it worth it?

File: lib/Data/Validate/Sanctions.pm [hunk 3]
```diff
@@ -40,6 +40,10 @@ sub get_sanction_file {
 }
 
 sub is_sanctioned {        ## no critic (RequireArgUnpacking)
+    return (get_sanctioned_info(@_))->{matched};
```
Comment (chylli) [inline lib/Data/Validate/Sanctions.pm]: why it need a pair of parenthesis around `get_sanctioned_info` ?

**Pull request merged**